hazelcast/src/main/java/com/hazelcast/multimap/impl/MultiMapService.java

```diff
@@ -322,7 +322,11 @@ public void insertMigratedData(int partitionId, Map<String, Map<Data, MultiMapVa
 
             for (Map.Entry<Data, MultiMapValue> multiMapValueEntry : collections.entrySet()) {
                 MultiMapValue multiMapValue = multiMapValueEntry.getValue();
-                container.getMultiMapValues().put(multiMapValueEntry.getKey(), multiMapValue);
+                MultiMapValue prevValue = container.getMultiMapValues().put(multiMapValueEntry.getKey(), multiMapValue);
```

Comment:
This approach suffers from lack of cohesion. So the logic for updating the size is now at multiple places.

I would expect to have a method on the container that knows how to deal with an inserted piece of data and update whatever state needs to be updated. Otherwise you end up with a very procedural approach (so hide the size increment stuff).